test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -10503,95 +10503,6 @@ TEST_P(XdsRbacTestWithActionPermutations, MethodPostPermissionAnyPrincipal) {
   SendRpc([this]() { return CreateInsecureChannel(); }, {}, {},
           /*test_expects_failure=*/GetParam().rbac_action() == RBAC_Action_DENY,
           grpc::StatusCode::PERMISSION_DENIED);
-  // Test an RPC with a different method type
```

Comment:
@yashykt, does this test still have any value without the ability to test other methods?

Reply:
Hmm, I guess we'll need testing (or some way to be confident) that all other method types are rejected. Otherwise, this test is not really needed.

Reply:
I want to make sure that we don't open ourselves to a security-related bug here.  [gRFC A41](https://github.com/grpc/proposal/blob/master/A41-xds-rbac.md) says:
> For this design, `headers` can include `:method`, `:authority`, and `:path` matchers and they should match the values received on-the-wire independent of whether they are stored in Metadata or in separate APIs. `:method` can be hard-coded to `POST` if unavailable and a code audit confirms the server denies requests for all other method types.

Do we have a guarantee that the transport will reject GET or PUT requests once this PR goes in?  If so, then that offers at least some assurance that the current behavior will be correct, even without the tests.  But it's really not the same as having a test.

Perhaps more importantly, if we re-add support for GET or PUT in the future, how will we remember to re-add these tests?

CC @ejona86 for his input as well.

Reply:
[http_server_filter.cc does this](https://github.com/grpc/grpc/pull/28922/files#diff-ecfb39bf6f4efc0bacabfded858c2346ef3b07c91babb8cbc5d4f286b28b70deL125)

Reply:
Okay.  But I think we should at least keep the test that shows that the rules does match when it specifies that the method is POST, since we know that we're going to hit that case.

For the negative case, how about having a rule that checks that the method is GET and showing that it does not match?

Reply:
Given that we will be removing the infrastructure to send requests with GET/PUT, this would mean we would need to handcraft those requests.

Reply:
I'm not proposing that we craft a different request -- all requests will continue to be POST.  I'm suggesting that we have one test with a rule that requires the method to be POST that does match, and another test with a rule that requires the method to be GET that does *not* match.

Reply:
Done